environs/config/config.go

```diff
@@ -1335,13 +1343,39 @@ func (c *Config) NumProvisionWorkers() int {
 	return value
 }
 
+// validateNumProvisionWorkers ensures the number cannot be set to
+// more than 100.
+// TODO: (hml) 26-Feb-2024
+// Once we can better link the controller config and the model config,
+// allow the max value to be set in the controller config.
+func (c *Config) validateNumProvisionWorkers() error {
+	value, ok := c.defined[NumProvisionWorkersKey].(int)
+	if ok && value > 100 {
+		return errors.Errorf("%s: must be less than 100", NumProvisionWorkersKey)
+	}
+	return nil
+}
+
 // NumContainerProvisionWorkers returns the number of container provisioner
 // workers to use.
 func (c *Config) NumContainerProvisionWorkers() int {
 	value, _ := c.defined[NumContainerProvisionWorkersKey].(int)
 	return value
 }
 
+// validateNumContainerProvisionWorkers ensures the number cannot be set to
+// more than 25.
+// TODO: (hml) 26-Feb-2024
+// Once we can better link the controller config and the model config,
+// allow the max value to be set in the controller config.
+func (c *Config) validateNumContainerProvisionWorkers() error {
+	value, ok := c.defined[NumContainerProvisionWorkersKey].(int)
+	if ok && value > 25 {
```

Comment:
same here, and then you'd want to include the const in the Errorf string as well. (`"%s: must be less than %d"`)